### ISSUE TITLE
[feature] Allow underscores at start of fileGroup identifiers

### ISSUE THREAD

**Issue by jameslawson** [moonrepo/moon#978]
**Is your feature request related to a problem? Please describe.**

- Right now it appears fileGroup identifiers can not start with one or two underscores, I see this error message:
![image](https://github.com/moonrepo/moon/assets/1913533/7c12f7da-3a43-477c-af6d-a7c4512cb430)

- I'd like underscores at the start of my filegroups in my repo so that when engineers are reading, they can quickly distinguish *inherited* fileGroups from *non-inherited* fileGroups in a succinct way via the file group name:

```yaml
# project's moon.yml
tasks:
  typecheck:
    command: 'tsc --build --pretty --verbose'
    inputs:
      - 'build'
      - '@files(__tsconfig)'
      - '@globs(__sources)'
```

**Describe the solution you'd like**

- I was hoping to setup a convention where in my monorepo, any fileGroups with the `__` prefix is inherited, and anything without the `__` prefix are defined in a local project. 

- `__tsconfig` is defined at workspace [`tasks.yml`](https://moonrepo.dev/docs/config/tasks#filegroups).  And similarly I'd have `__typescript` for inherited typescript fileGroup and `__babel` for inherit babel config files and so on. Projects would continue to define fileGroups without underscores e.g. `testFiles`, `frontendAssets`, `apiDocs`, etc... . 

- Allowing underscores can allow for more flexible names for other teams like:  `@globs(__deprecated__sources)`, `@globs(__do_not_use__sources)` for large monorepos where that team is struggling to hit the ideal situation of refactoring all projects at once, and instead need to change fileGroups incrementally to mitigate risk. 

**Describe alternatives you've considered**

- Identifiers `workspace_typescript`, `wp__typescript`, `root__typescript` and `typescript__`  work but  feel a little bit clunky.  Just writing `__typescript` is short, yet clearly calls out that it's different to `typescript`.

Wondering how easy feature is easy to add in, and if other projects would get value from this? Are the rules on identifiers format strong quite firmly in place rules or still fairly open for changes? 

Thank you for Moon, really enjoying using it so far!

**Comment by milesj** [comment]
Sounds reasonable. Will add for the next release!

**Comment by jameslawson** [comment]
Yay!! Thank you so much 🙌

**Comment by milesj** [comment]
Landed in v1.11 https://moonrepo.dev/blog/moon-v1.11#unlocked-features
